app/routes/goal_routes.py

```diff
@@ -1 +1,142 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+from datetime import datetime
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+@goals_bp.post("")
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+
+    except KeyError:
+        response = {"details": "Invalid data"}
```

Comment:
It would be nice if we could provide more details about the invalid request body so that the client sending the request knows what about the body to fix.

```suggestion
    except KeyError as error:
        response = {"details": f"Invalid data - missing {error.args[0]}"}
```